### PR TITLE
feat: array of slots

### PR DESCRIPTION
**Pull request by lmiller1990**
Continue from: https://github.com/vuejs/vue-test-utils-next/pull/573

resolves https://github.com/vuejs/vue-test-utils-next/issues/482

Now you may pass an array of slots. There is some TS weirdness, I have no idea if the types are wrong or my code is wrong, but everything works as expected.

### CODE REVIEW

File: src/mount.ts [hunk 1]
```diff
@@ -269,33 +299,24 @@ export function mount(
...
-        if (typeof slot === 'object' && 'render' in slot && slot.render) {
-          acc[name] = slot.render
-          return acc
-        }
+        if (Array.isArray(slot)) {
+          const normalized = slot.reduce<Array<Function | VNode>>(
+            (acc, curr) => {
+              const toF = slotToFunction(curr)
+              if (isObject(curr) && 'render' in curr) {
+                const rendered = h(toF as any)
```
Comment (lmiller1990) [inline src/mount.ts:307]: Not entirely clear why I need to pass `toF` (short for `toFunction`) into `h` here... but you need to.

File: src/mount.ts [hunk 2]
```diff
@@ -261,6 +264,33 @@ export function mount(
...
+      return slot.render
+    }
+
+    if (typeof slot === 'function') {
+      return slot
+    }
+
+    if (typeof slot === 'object') {
+      return () => slot
+    }
```
Comment (cexbrayat) [inline src/mount.ts:278]: nit: I would write a single ` if (typeof slot === 'object')` and then deal with two cases inside.

File: src/mount.ts [hunk 3]
```diff
@@ -261,6 +264,33 @@ export function mount(
     to.appendChild(el)
   }
 
+  function slotToFunction(slot: Slot): Function {
```
Comment (cexbrayat) [inline src/mount.ts:267]: nit: Function is rarely used as a type. We usually prefer `() => string` for example.

File: src/mount.ts [hunk 4]
```diff
@@ -269,33 +299,24 @@ export function mount(
...
-        // case of an SFC getting passed
-        if (typeof slot === 'object' && 'render' in slot && slot.render) {
-          acc[name] = slot.render
-          return acc
-        }
+        if (Array.isArray(slot)) {
+          const normalized = slot.reduce<Array<Function | VNode>>(
+            (acc, curr) => {
+              const toF = slotToFunction(curr)
+              if (isObject(curr) && 'render' in curr) {
```
Comment (cexbrayat) [inline src/mount.ts:306]: I have a hard time understanding the code. Maybe some comments would help. `toF` could probably be renamed to something more meaningful (`slotAsFunction`?).

It feels like this check is somewhat redundant with what is done in `slotToFunction`. I don't know exactly how, but it looks like this should be refactor/simplified.

File: src/mount.ts (hunk 1 above)
Comment (cexbrayat) [reply src/mount.ts:307]: This is indeed weird. I don't know enough about slot internal behavior to help sadly.

File: src/mount.ts (hunk 4 above)
Comment (lmiller1990) [reply src/mount.ts:306]: Naming definitely can be improved.

I also have a hard time understanding this, or why it works - the amount of ways you can create a Vue component (and as such, a lot) is a bit overwhelming. I can think of at least 5 different ways:

- class component
- function component
- SFC
- setup returning render function
- string (`i am a string`)
- template (`<div>template</div>`)

Throw in scoped slots, and you end up with this.

File: src/mount.ts (hunk 3 above)
Comment (lmiller1990) [reply src/mount.ts:267]: Ok, I will change this. I suppose it'll be like `type SlotLike = () => string | (() => string) | { render: () => string }` or something to that meaning.

Comment (lmiller1990) [reply src/mount.ts:267]: The type here is *very* complicated, it wants me to specify some very large union of public component instance, functional components... I just removed the entire return type and will let TS figure it out.

File: src/mount.ts (hunk 4 above)
Comment (lmiller1990) [reply src/mount.ts:306]: DOne

**Pull request merged**